### PR TITLE
Always do unfold headers to fix long header test match

### PR DESCRIPTION
**Pull request by danieledepetrini**
This patch will fix (at least) header test with "matches" criteria in case of long (folded) headers

### CODE REVIEW

File: util/src/main/java/org/apache/jsieve/util/check/ScriptCheckMailAdapter.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
 import javax.mail.MessagingException;
 import java.io.IOException;
 import java.util.*;
+import javax.mail.internet.MimeUtility;
```
Comment (mbaechler) [inline util/src/main/java/org/apache/jsieve/util/check/ScriptCheckMailAdapter.java]: we usually keep import sorted

File: util/src/main/java/org/apache/jsieve/util/check/ScriptCheckMailAdapter.java [hunk 2]
```diff
@@ -137,7 +138,10 @@ public void reset() {
             try {
                 String[] values = mail.getHeader(name);
                 if (values != null) {
-                    result = Arrays.asList(values);
+                    //We need to do unfold headers here
+                    result = new LinkedList<String>();
+                    for (String value: values)
```
Comment (mbaechler) [inline util/src/main/java/org/apache/jsieve/util/check/ScriptCheckMailAdapter.java]: could you just add curly braces for this code block ?

**Comment by danieledepetrini** [comment]
Added unit test and fixed code style, is that OK now ?

**Comment by danieledepetrini** [comment]
Great! So what's going to happen next ?

**Comment by mbaechler** [comment]
Usually, we wait for about a day for other people to review the PR is they
want to.
After that, integration on master.
Can you live with a SNAPSHOT version ?

**Comment by danieledepetrini** [comment]
Sure NP, I'm on my private branch since March so I can survive for a few more days :-)

**Comment by Laura-Royet** [comment]
See: https://github.com/apache/james-jsieve/pull/15

File: util/src/test/java/org/apache/jsieve/util/HeaderMatchingTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
+package org.apache.jsieve.util;
```
Comment (chibenwa) [inline util/src/test/java/org/apache/jsieve/util/HeaderMatchingTest.java]: License

**Comment by chibenwa** [comment]
Hi, I just merged the issue on master.

Could you please close the issue ?

**Pull request closed without merging**